### PR TITLE
Expand category icon

### PR DESCRIPTION
**Pull request by brianteeman**
Continuing the work to ensure that all uses of font icons are accessible and following the advice of Font Awesome (http://fontawesome.io/accessibility/) this PR ensures that the interactive icons used in category views in the front end to expand or collapse the display of subcategories have both an aria-label on the link and aria-hidden on the icon.

Note the code is different to other related PR as the icon is changed with JS

### Testing
The easiest way to test is with the tesing sample data installed and then the "contact categories" and the "article categories" menu items - see video
plus

### CODE REVIEW

File: components/com_contact/views/categories/tmpl/default.php [hunk 1]
```diff
@@ -19,6 +24,12 @@
 		btn.on('click', function() {
 			btn.find('span').toggleClass('icon-plus');
 			btn.find('span').toggleClass('icon-minus');
+			if (btn.attr('aria-label') == Joomla.JText._('JGLOBAL_EXPAND'))
```
Comment (C-Lodder) [inline components/com_contact/views/categories/tmpl/default.php]: `===`

File: components/com_content/views/categories/tmpl/default.php [hunk 2]
```diff
@@ -19,6 +24,12 @@
 		btn.on('click', function() {
 			btn.find('span').toggleClass('icon-plus');
 			btn.find('span').toggleClass('icon-minus');
+			if (btn.attr('aria-label') == Joomla.JText._('JGLOBAL_EXPAND'))
```
Comment (C-Lodder) [inline components/com_content/views/categories/tmpl/default.php]: `===`

**Comment by C-Lodder** [comment]
`JGLOBAL_COLLAPSE="Show less categories"` are you should you want to go with that? What if it's not a category? Perhaps just

```
JGLOBAL_COLLAPSE="Show less"
```
?

**Comment by brianteeman** [comment]
@C-Lodder it is only used for categories

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 5af6c86cab889e6e76696f65d6d0f75d4c7cf539<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18459">issues.joomla.org/tracker/joomla-cms/18459</a>.</sub>

**Comment by C-Lodder** [comment]
@brianteeman - I was thinking more along the lines of, if we want to use it for 3rd party extensions or the collapse custom element in J4

**Comment by brianteeman** [comment]
@C-Lodder for accessibility the text should be meaningful so it needs the "category" part so you know what you are showing more of. I think what is confusing right now is the key that I have used for the PR as it is very generic and not specific to categories. As I stated in the original post I am seeking guidance from @infograf768 for the correct key to use and these keys are just for testing purposes. For your use case in J4 we can have different strings possibly with a sprintf

**Comment by dgrammatiko** [comment]
@brianteeman I think @C-Lodder is right, correcting thins per view is gonna be a lot of repeated code instead of using a global collapse custom element specific to joomla's needs. That said this PR is against staging, so no CE thus it's ok the way you're doing it, but for J4 we need to use Custom Element for such behaviours

**Comment by brianteeman** [comment]
Do what you want for J4 - this is for J3. I have given up looking at j4 as all development has pretty much stopped while waiting for ux and a11y team

**Comment by dgrammatiko** [comment]
>  as all development has pretty much stopped while waiting for ux and a11y team

Sadly that's true

**Comment by infograf768** [comment]
I these new strings are strictly related to categories, I suggest to change the constants to
`JGLOBAL_EXPAND_CATEGORIES` and `JGLOBAL_COLLAPSE_CATEGORIES`

**Comment by brianteeman** [comment]
Thanks @infograf768 i will update the pr with those keys

**Comment by C-Lodder** [comment]
Thanks @brianteeman

**Comment by C-Lodder** [comment]
I have tested this item :white_check_mark: successfully on 891fa71017e7550f4e9a0574c60f79f54e43875c<br /><br />Based on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18459">issues.joomla.org/tracker/joomla-cms/18459</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 891fa71017e7550f4e9a0574c60f79f54e43875c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18459">issues.joomla.org/tracker/joomla-cms/18459</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**